### PR TITLE
add icon build

### PR DESCRIPTION
**Pull request by Julien-laville**

### CODE REVIEW

**Comment by nlehuby** [comment]
Here are a few tests to add:
* :x:  is there at least a layer in the style named `poi-level-*` ?
* :x: is there a `icons.yml` file in the directory ?
* for each elem in the yml file:
   * :warning:  is the color listed in the file?
   * :warning: is the icon listed in the file ?
   * :x:  is the icon present in the icon directory ?

File: task/build_cli.js [hunk 1]
```diff
@@ -30,6 +30,14 @@ render the labels. If not defined, 'name' tag is always used.",
...
     }
   })
+  .options({
+    webfont: {
+      describe: "Enable the build process of an icon font containing style icons",
+      type: 'boolean',
+      nargs: 1,
+      default: false
+    }
+  })
```
Comment (nlehuby) [inline task/build_cli.js]: the `icons` arg is also needed

File: task/icons/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        if (icon.color) {
+          colorsCases.push(["==",
+            ["get", "subclass"],
+            icon.subclass])
+          colorsCases.push(icon.color)
+        }
+      }
+    })
+
+    iconsCases.push('default')
```
Comment (nlehuby) [inline task/icons/index.js]: For now, the default icon should be `star-stroked-11`

File: task/build_cli.js (hunk 1 above)
Comment (Julien-laville) [reply task/build_cli.js]: done (je crois)

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const path = require('path');
+
+const styleTest = require('./style')
+const layerTest = require('./layers')
+const fontTest = require('./fonts')
+const i18nTest = require('./i18n')
```
Comment (nlehuby) [inline test/index.js:6]: this one does not exist for now :cry:

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const iconsTest = require('./icons')
+
+let args = require('./../lib/style_args').argv;
```
Comment (nlehuby) [inline test/index.js:9]: this one does not exist either :sob:

File: task/build_cli.js [hunk 5]
```diff
@@ -1,61 +1,16 @@
...
-const confStr = fs.readFileSync(args.conf, 'utf8')
-const styleStr = fs.readFileSync(stylePath, 'utf8')
-const style = JSON.parse(styleStr)
-const jsonconf = JSON.parse(confStr)
+const fs = require('fs-extra');
+const path = require('path');
+const build = require('../lib/build');
+const mkdirp = require('mkdirp');
+
+let args = require('./../lib/style_args').argv;
```
Comment (nlehuby) [inline task/build_cli.js]: this one does not exist either :disappointed:

**Review by nlehuby — CHANGES_REQUESTED** [review]
:crying_cat_face:

**Review by nlehuby — APPROVED** [review]
:+1:

**Review by amatissart — APPROVED** [review]

**Pull request merged**